src/libasr/codegen/asr_to_fortran.cpp

```diff
@@ -870,7 +870,20 @@ class ASRToFortranVisitor : public ASR::BaseVisitor<ASRToFortranVisitor>
 
     // void visit_ArraySection(const ASR::ArraySection_t &x) {}
 
-    // void visit_ArraySize(const ASR::ArraySize_t &x) {}
+    void visit_ArraySize(const ASR::ArraySize_t &x) {
+        visit_expr(*x.m_v);
+        std::string r = "size(" + s;
+        if (x.m_dim) {
+            r += ", ";
+            int dim = down_cast<ASR::Integer_t>(x.m_dim)->m_kind;
+            for(int i = 0; i < dim; i++) {
+                visit_expr(*x.m_dim);
+                r += s;
+                if (i < dim-1) r += ", ";
+            }
```

Comment:
ASR::Integer_t is a `ttype` node not `expr` node (You can confirm this in ASR.asdl file).
You can analyze ASR output using --show-asr to handle this.

From ASR of arrays_01_size.f90 you can see it has `IntegerConstant`, so you have to do:
```suggestion
            visit_expr(*x.m_dim);
            r += s;
```

Also, x.m_dim can be IntegerConstant or Var. Later we have to add a check for Var as well

Reply:
Also, you have to visit m_kind as well.

Reply:
I don't think you can do `visit_expr(*x.m_dim);`, can you? I think the original code is needed, but I can see a mistake there, it should have `visit_expr(*x.m_dim[i]);`.

Reply:
```
    | ArraySize(expr v, expr? dim, ttype type, expr? value)
...
```
The dim is optional and scalar here, so I think `visit_expr(*x.m_dim)` would be enough.